test/parser/parse.test.ts

```diff
@@ -1855,3 +1857,42 @@ See more help with --help`)
     })
   })
 })
+
+describe('stdin', () => {
+  let stdin: ReturnType<typeof mockStdin.stdin>
+  let out: ParserOutput<
+    {
+      myflag: string | undefined
+    },
+    OutputFlags<any>,
+    OutputArgs<any>
+  >
+  const stdinValue = 'x'
+
+  const execute = async (): Promise<void> => {
+    out = await parse(['--myflag', '-'], {
+      flags: {
+        myflag: Flags.string({allowStdin: true}),
+      },
+    })
+  }
+
+  beforeEach(() => {
+    stdin = mockStdin.stdin()
+  })
+
+  describe('allow stdin', () => {
+    beforeEach(() => {
+      execute()
+      stdin.send(stdinValue)
+      stdin.end()
+    })
+
+    it('should read stdin as input for command', () => {
+      afterEach(() => {
```

Comment:
Why did u wrap these tests with afterEach?

Reply:
I ran into some issues with the asserting executing before the stdin was read in, due to the parsing method being asynchronous. This is how I was able to force it to wait.

Reply:
Does making the `it` callback `async` help this? This is what I have in mind:

```typescript
describe('allowStdin', () => {
  it('should read stdin as input for command', async () => {
    const stdin = mockStdin.stdin()
    stdin.send(stdinValue)
    stdin.end()
    const out = await execute()
    expect(out.raw[0].input).to.equal(stdinValue)
    expect(out.flags.myflag).to.equals(stdinValue)
  }
})
```

I didn't test that by the way. Just how I would attempt to do it myself.

You could also consider mocking the `readStdin` function in `src/parser/parse.ts` instead of trying to mock `stdin` directly

Reply:
I just tried this. I still can't get it to work without using `afterEach()`. Depending on what I try it will either:

1. Throw an error that `out.raw is undefined` because it is still waiting for `readStdin()` to finish
2. Fail the assertion `expected '-' to equal 'x'` because something about making the `it` callback async is causing it to **not** want to wait for `readStdin()` to finish

Reply:
I'm happy to try anything else though, let me know if the current approach is a major concern or bad practice.

Reply:
My main concern is that it's hiding some sort of race condition that'll come back to bite us later. If I were writing this I would stub `readStdin` instead of trying to use `mock-stdin`. Something like this:

```typescript
import {SinonSandbox, createSandbox} from 'sinon'
import * as parser from '../../src/parser'

describe('allowStdin', () => {
  let sandbox: SinonSandbox

  beforeEach(() => {
    sandbox = createSandbox()
  })

  afterEach(() => {
    sandbox.restore()
  })

  it('should read stdin as input for flag', async () => {
    sandbox.stub(parser, 'readStdin', 'x')
    const result = await parse(['--myflag', '-'], {
      flags: {
        myflag: Flags.string({allowStdin: true}),
      },
    })

   // assertions here
  })
})
```

Reply:
@mdonnalley How does that look now?